v12/storage-dotnet-redundancy-ha-ra-grs/Program.cs

```diff
@@ -0,0 +1,262 @@
+﻿using System.Diagnostics.Tracing;
+using System.IO;
+using Azure;
+using Azure.Core;
+using Azure.Core.Diagnostics;
+using Azure.Core.Pipeline;
+using Azure.Identity;
+using Azure.Storage;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+
+namespace StorageRedundancy
+{
+    internal class ShowRequestInfoPolicy : HttpPipelinePolicy
+    {
+        public override void Process(HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
+        {
+            string? host = message.Request.Uri.Host;
+            if (host is not null) 
+            {
+                string region = host.Contains("-secondary") ? "secondary" : "primary";
+                Console.WriteLine($"Request sent to {region} region endpoint");
+            }
+            
+            ProcessNext(message, pipeline);
+        }
+
+        public override async ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
+        {
+            string? host = message.Request.Uri.Host;
+            
+            if (host is not null)
+            {
+                string region = host.Contains("-secondary") ? "secondary" : "primary";
+                Console.WriteLine($"Request sent to {region} region endpoint");
+            }
+            await ProcessNextAsync(message, pipeline);
+
+            
+        }
+    }
+
+    public class Program
+    {
+        static int retryCount = 0;
+        const int retryThreshold = 3;
+        static int secondaryReadCount = 0;
+        const int secondaryReadThreshold = 5;
+        static void Main(string[] args)
+        {
+            Console.WriteLine("Azure Storage redundancy sample\n ");
+
+            try
+            {
+                RunStorageRedundancyAsync().GetAwaiter().GetResult();
```

Comment:
```suggestion
                await RunStorageRedundancyAsync();
```

C# has now supported `Task` returning `Main` methods for a while, allowing for `async/await`. Try to avoid `.GetAwaiter().GetResult()`.